### PR TITLE
Remove lodash from MonthPicker and buildDataAttributes

### PR DESCRIPTION
**Pull request by mattsjones**
Other PR tweaks
- Add some tests for MonthPicker
- Remove Dropdown child validation

### CODE REVIEW

File: lib/components/private/buildDataAttributes.ts [hunk 1]
```diff
@@ -1,6 +1,9 @@
-import mapKeys from 'lodash/mapKeys';
-
 export type DataAttributeMap = Record<string, string | number>;
 
-export default (data: DataAttributeMap = {}) =>
-  mapKeys(data, (_, attrName) => `data-${attrName}`);
+export default (
+  data: DataAttributeMap = {},
+): { [dataAttribute: string]: string } =>
+  Object.assign(
```
Comment (michaeltaranto) [inline lib/components/private/buildDataAttributes.ts:6]: Any risk that consumers might not have polyfilled this? Should we add this in the release notes?

File: .changeset/pink-jokes-talk.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+'braid-design-system': patch
+---
+
+Drop lodash usage to decrease bundle size
+
+Affected components:
+
+- All field components
```
Comment (michaeltaranto) [inline .changeset/pink-jokes-talk.md:9]: Maybe capture that this is only relating to use of the `data` prop.

File: lib/components/private/buildDataAttributes.ts (hunk 1 above)
Comment (mattsjones) [reply lib/components/private/buildDataAttributes.ts:6]: Good question. How about I switch to `object-assign` which react uses internally?

File: .changeset/pink-jokes-talk.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+- All Menu components
```
Comment (michaeltaranto) [inline .changeset/pink-jokes-talk.md:10]: For ease of scanning, i'd call out names here, `OverflowMenu` and `MenuRenderer`, and also maybe call out `MonthPicker` as a separate line item.

File: lib/components/private/buildDataAttributes.ts (hunk 1 above)
Comment (michaeltaranto) [reply lib/components/private/buildDataAttributes.ts:6]: Either that, or just spread into a new literal maybe? Given we are spreading below.

Comment (mattsjones) [reply lib/components/private/buildDataAttributes.ts]: Changed to a `for of` implementation

**Pull request merged**